test/test-parse-function.cpp

```diff
@@ -1445,6 +1445,16 @@ TEST_F(Test_Parse_Function, incomplete_function_body) {
   }
 }
 
+TEST_F(Test_Parse_Function, report_unclosed_code_block_based_on_indentation) {
+  {
+    test_parse_and_visit_statement(
+        u8"function f() {\n\tif (true) {\n\t\tbody();\n}"_sv,  //
+        u8"             ^ Diag_Unclosed_Code_Block"_diag,
+        u8"                            ^ Diag_Mismatched_Curly_Indentation.left_curly\n"_diag  //
+        u8"                                            ^ .right_curly"_diag);
```

Comment:
For this example, I think the best experience would be to report something like this instead:
```suggestion
    test_parse_and_visit_statement(
        u8"function f() {\n\tif (true) {\n\t\tbody();\n}"_sv,  //
        u8"                                            ` Diag_Unclosed_Code_Block_V2.expected_right_curly\n"_diag,
        u8"                            ^ Diag_Unclosed_Code_Block_V2.left_curly\n"_diag);
```
We would report only one diagnostic and it would tell the user how to fix the problem ("missing '}'", "matching '{' here"). This was my intent with #282.

However, if we cannot get such a good diagnostic, then what you have is fine.